src/opm/io/eclipse/ESmry.cpp

```diff
@@ -496,6 +519,48 @@ std::string ESmry::makeKeyString(const std::string& keywordArg, const std::strin
     return keyStr;
 }
 
+std::string ESmry::unpackNumber(const SummaryNode& node) const {
+    if (node.category == SummaryNode::Category::Block ||
+        node.category == SummaryNode::Category::Connection) {
+        int _i,_j,_k;
+        ijk_from_global_index(node.number, _i, _j, _k);
+
+        return std::to_string(_i) + "," + std::to_string(_j) + "," + std::to_string(_k);
+    } else if (node.category == SummaryNode::Category::Region && node.keyword[2] == 'F') {
+        // NUMS = R1 + 32768*(R2 + 10)
+        int r2 = 0;
+        int y = 32768 * (r2 + 10) - node.number;
+
+        while (y < 0) {
+            r2++;
+            y = 32768 * (r2 + 10) - node.number;
+        }
+
+        r2--;
+
+        const int r1 = node.number - 32768 * (r2 + 10);
+
+        return std::to_string(r1) + "-" + std::to_string(r2);
```

Comment:
Wow, that's a complicated way of writing
```C++
const auto r1 =  node.number % (1 << 15);
const auto r2 = (node.number / (1 << 15)) - 10;
```

Reply:
> Wow, that's a complicated way of writing
> 
> ```c++
> const auto r1 =  node.number % (1 << 15);
> const auto r2 = (node.number / (1 << 15)) - 10;
> ```

I agree, updated. (Code was originally adapted from code in `makeKeyString`, extracted as-is.)